go/consensus/cometbft/apps/roothash/state/state_test.go

```diff
@@ -116,11 +116,66 @@ func TestSeparateRuntimeRoots(t *testing.T) {
 	require := require.New(t)
 
 	appState := abciAPI.NewMockApplicationState(&abciAPI.MockApplicationStateConfig{})
-	ctx := appState.NewContext(abciAPI.ContextBeginBlock)
-	defer ctx.Close()
 
+	ctx := appState.NewContext(abciAPI.ContextInitChain)
 	st := NewMutableState(ctx.State())
+	err := st.SetConsensusParameters(ctx, &api.ConsensusParameters{})
+	require.NoError(err, "SetConsensusParameters")
+	ctx.Close()
+
+	ctx = appState.NewContext(abciAPI.ContextBeginBlock)
+	defer ctx.Close()
+
+	st = NewMutableState(ctx.State())
 
+	var runtime registry.Runtime
+	err = runtime.ID.UnmarshalHex("8000000000000000000000000000000000000000000000000000000000000000")
+	require.NoError(err, "UnmarshalHex")
+
+	blk := block.NewGenesisBlock(runtime.ID, 0)
+	err = blk.Header.StateRoot.UnmarshalHex("0000000000000000000000000000000000000000000000000000000000000001")
+	require.NoError(err, "UnmarshalHex")
+	err = blk.Header.IORoot.UnmarshalHex("0000000000000000000000000000000000000000000000000000000000000002")
+	require.NoError(err, "UnmarshalHex")
+	err = st.SetRuntimeState(ctx, &api.RuntimeState{
+		Runtime:            &runtime,
+		GenesisBlock:       blk,
+		CurrentBlock:       blk,
+		CurrentBlockHeight: 1,
+		LastNormalRound:    0,
+		LastNormalHeight:   1,
+	})
+	require.NoError(err, "SetRuntimeState")
+
+	stateRoot, err := st.StateRoot(ctx, runtime.ID)
+	require.NoError(err, "StateRoot")
+	require.EqualValues(blk.Header.StateRoot, stateRoot)
+
+	ioRoot, err := st.IORoot(ctx, runtime.ID)
+	require.NoError(err, "IORoot")
+	require.EqualValues(blk.Header.IORoot, ioRoot)
+}
+
+func TestPastRuntimeRoots(t *testing.T) {
```

Comment:
Unit test for shrink should be in this file.

Reply:
It cannot, since the change parameters call can only be called from outside the `state` package.

Reply:
What you are testing is that `ChangeParameters` correctly shrinks roots by calling `ShrinkPastRoots`. I would unit test the latter method alone, e.g. by inserting few roots and check if some are removed.

Reply:
Done, please take a look :)